### ISSUE TITLE
kernel panic after upgrading to 4.12.4-1.e17.elrepo.x86_64 (hyper-V VM)

### ISSUE THREAD

**Issue by jallumuk** [rockstor/rockstor-core#1886]
Running Rockstor as a Hyper-v VM under Win10Pro with guest services active and HDD pass-thru for main storage HDD (single HDD), VHDX for Rockstor OS.

After upgrading to stable release 4.12.4-1.e17.elrepo.x86_64 from 4.10.6-1 (image), I receive a kernel panic (image) and Rockstor crashes out. Rockstor starts up fine for the first few minutes then as soon as the load picks up, Docker Resilio Sync was running at the time, it the VM crashes.
Rockstor Boot Menu:
![rockstor-startup-hyper-v](https://user-images.githubusercontent.com/35896810/35484485-41c19734-0448-11e8-889c-60fa9d4536ac.jpg)
Rockstor VM Crash console:
![rockstor-sync-hyper-v-crash](https://user-images.githubusercontent.com/35896810/35484487-46ecca3a-0448-11e8-93ac-edd9959c4c11.jpg)
Rockstor VM Settings:
![image](https://user-images.githubusercontent.com/35896810/35484519-d4c53716-0448-11e8-909a-0387d9b5f3d9.png)

After a little research, I found this recent CentOS incident: https://bugs.centos.org/view.php?id=12672&nbn=6
I am currently selecting the previous release at boot to enable me to run successfully.

**Comment by phillxnet** [comment]
@jallumuk Thanks for the well presented report. However as we used an unmodified kernel-ml from elrepo:
http://elrepo.org/tiki/tiki-index.php?page=kernel-ml
it may make more sense to report your finding there, although our kernel is a fair bit behind their latest offering currently. But all the same they may have seen this previously and be able to advise you accordingly. Another option is to try installing one of their newer kernels: such as is suggested to address the Meltdown and Spectre issues (at least in part) in the following forum thread:
https://forum.rockstor.com/t/what-to-do-about-meltdown-and-spectre/4296
ie an elrepo 4.14.13.
You will similarly have to select it on boot but if that works for you then you can then ask on the forum how to make that the default.

So in short it's an upstream issue as we make no changes to their kernels and so they (elrepo folks) would be the ones to advise you on this issue.

Let us know how you get on.

**Comment by jallumuk** [comment]
Hi Phil
I finally manged to get some time to come back to this and am now running what appears to be a stable kernel. I decided to install kernel-ml-4.15.0-1.el7.elrepo.x86_64.rpm which is the latest stable elrepo kernel. To do this from the shell under root,  I had to:
rpm --import https://www.elrepo.org/RPM-GPG-KEY-elrepo.org
rpm -i kernel-ml-4.15.0-1.el7.elrepo.x86_64.rpm

I now have 3 kernels available, the original working kernel-ml-4.10.6-1.el7.elrepo.x86_64, the one that crashes after the last Rockstor stable update, kernel-ml-4.12.4-1.el7.elrepo.x86_64 and the latest one kernel-ml-4.15.0-1.el7.elrepo.x86_64.rpm that seems to run without any problems:
![2018-02-03](https://user-images.githubusercontent.com/35896810/35766997-0f28c5c6-08db-11e8-85d6-436e94e1abfc.png)

Currently, I am selecting the latest kernel-ml-4.15.0-1.el7.elrepo.x86_64.rpm manually, but I will research this next to make this the default boot kernel.

My worry now is I am out of kilter with the official Rockstor update program and now get the message:
![2018-02-03 1](https://user-images.githubusercontent.com/35896810/35767026-b3a65f82-08db-11e8-884d-5b3d54cf114d.png)

Do you forsee any other way around my predicament and will I still receive notification of Rockstor updates.

Thanks

**Comment by jallumuk** [comment]
I should have added I downloaded and copied the kernel-ml-4.15.0-1.el7.elrepo.x86_64.rpm file to a Rockstor share first and installed from there.

**Comment by phillxnet** [comment]
@jallumuk Thanks for the update and excellent feedback. Glad your up and running again.

I'd like to ask that you start a forum post on your findings / adventure:
https://forum.rockstor.com/
as that is a more appropriate medium for a discussion on this manual configuration 'work around'. Plus it will receive more eyes and my / others answer to you questions would be more usefully / appropriate there as we have already had some discussions of this sort of mod (newer kernels) but not for a while, so your experience / report will help to update that info on the forum for others. A new thread should do it and it may well help those looking to mitigate the recent upstream kernel issues re meltdown / spectra referenced earlier (at least until we update our issued kernel). You could cut and paste your main 4.15 related post from this issue and link to this issue for context.

If you think it's appropriate you could leave this issue open until we update the 'included' kernel to this or a later version given your 'proof of fix' at least in your documented scenario.

If you don't fancy partaking in the forum let me know and I can answer here if need be.

**Comment by jallumuk** [comment]
Hi Phil
I raised a "HOW-TO" on the forum and hope this is the correct way, you can check it out at.
https://forum.rockstor.com/t/upgrading-to-latest-elrepo-kernel-kernel-ml-4-15-0-1-el7-elrepo-x86-64-rpm/4410
Many Thanks.
